objectivec/ort_enums.mm

```diff
@@ -119,9 +120,11 @@ ORTTensorElementDataType CAPIToPublicTensorElementType(ONNXTensorElementDataType
 size_t SizeOfCAPITensorElementType(ONNXTensorElementDataType capi_type) {
   return SelectAndTransform(
       kElementTypeInfos,
-      [capi_type](const auto& type_info) { return type_info.capi_type == capi_type; },
-      [](const auto& type_info) { return type_info.element_size; },
-      "unsupported tensor element type");
+      [capi_type](const auto& type_info) {
+        return type_info.element_size.has_value() && type_info.capi_type == capi_type;
+      },
+      [](const auto& type_info) { return type_info.element_size.value(); },
```

Comment:
we already did the check in the select fn, can avoid the extra check in optional::value.
```suggestion
      [](const auto& type_info) { return *type_info.element_size; },
```